app/domain/authentication/authn_oidc/update_input_with_username_from_id_token.rb

```diff
@@ -36,13 +35,13 @@ def validate_service_id_exists
         raise Errors::Authentication::AuthnOidc::ServiceIdMissing unless service_id
       end
 
-      def validate_credentials_include_id_token
-        id_token_field_name = "id_token"
-
-        # check that id token field exists and has some value
-        if decoded_credentials.fetch(id_token_field_name, "") == ""
-          raise Errors::Authentication::RequestBody::MissingRequestParam, id_token_field_name
+      def authorization_header_token(headers)
+        @id_token = if headers&.key?("HTTP_AUTHORIZATION")
```

Comment:
The `@` denotes an instance variable, which means it's available outside the method. Unless we need `@id_token` outside this method, please use the local variable `id_token`.

Reply:
fixed